services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -15,514 +15,415 @@
 import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.core.SdkBytes;
 import software.amazon.awssdk.core.SdkNumber;
-import software.amazon.awssdk.core.document.Document;
 import software.amazon.awssdk.enhanced.dynamodb.AttributeConverter;
 import software.amazon.awssdk.enhanced.dynamodb.AttributeConverterProvider;
-import software.amazon.awssdk.enhanced.dynamodb.DefaultAttributeConverterProvider;
 import software.amazon.awssdk.enhanced.dynamodb.EnhancedType;
 import software.amazon.awssdk.enhanced.dynamodb.document.EnhancedDocument;
 import software.amazon.awssdk.enhanced.dynamodb.internal.converter.ChainConverterProvider;
+import software.amazon.awssdk.enhanced.dynamodb.internal.converter.StringConverterProvider;
 import software.amazon.awssdk.enhanced.dynamodb.internal.converter.attribute.JsonItemAttributeConverter;
-import software.amazon.awssdk.protocols.json.internal.unmarshall.document.DocumentUnmarshaller;
+import software.amazon.awssdk.enhanced.dynamodb.internal.converter.attribute.ListAttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.internal.converter.attribute.MapAttributeConverter;
 import software.amazon.awssdk.protocols.jsoncore.JsonNode;
 import software.amazon.awssdk.protocols.jsoncore.JsonNodeParser;
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.utils.Lazy;
 import software.amazon.awssdk.utils.StringUtils;
 import software.amazon.awssdk.utils.Validate;
 
 /**
- * Default implementation of {@link EnhancedDocument}. This class is used by SDK to create Enhanced Documents.
- * Internally saves attributes in an attributeValueMap which can be written to DynamoDB without further conversion.
- * The attribute values are retrieved by converting attributeValue from attributeValueMap at the time of get.
+ * Default implementation of {@link EnhancedDocument} used by the SDK to create Enhanced Documents. Attributes are initially saved
+ * as a String-Object Map when documents are created using the builder. Conversion to an AttributeValueMap is done lazily when
+ * values are accessed. When the document is retrieved from DynamoDB, the AttributeValueMap is internally saved as the attribute
+ * value map. Custom objects or collections are saved in the enhancedTypeMap to preserve the generic class information. Note that
+ * no default ConverterProviders are assigned, so ConverterProviders must be passed in the builder when creating enhanced
+ * documents.
  */
 @Immutable
 @SdkInternalApi
 public class DefaultEnhancedDocument implements EnhancedDocument {
 
+    public static final IllegalStateException NULL_SET_ERROR = new IllegalStateException("Set must not have null values.");
+    private static final JsonItemAttributeConverter JSON_ATTRIBUTE_CONVERTER = JsonItemAttributeConverter.create();
     private static final JsonItemAttributeConverter JSON_ITEM_ATTRIBUTE_CONVERTER = JsonItemAttributeConverter.create();
```

Comment:
Do you need 2?

Reply:
Nope , this got added while merging from different branches